src/components/PeopleFilters.tsx

```diff
@@ -1,25 +1,141 @@
-export const PeopleFilters = () => {
+/* eslint-disable prettier/prettier */
+import { Person } from '../types';
+import classNames from 'classnames';
+import { SearchParams } from '../utils/searchHelper';
+import { SearchLink } from './SearchLink';
+import { useEffect } from 'react';
+
+type Props = {
+  people: Person[];
+  setCurrentPeopleList: (filteredPeople: Person[]) => void;
+  searchParams: URLSearchParams;
+  setSearchWith: (params: SearchParams) => void;
+};
+
+enum Sex {
+  All = '',
+  Male = 'm',
+  Female = 'f',
+}
+
+const createCenturieRange = (start: number, end: number) => {
+  const result = [];
+
+  for (let i = start; i <= end; i++) {
+    result.push(i);
+  }
+
+  return result;
+};
+
+const centurieRange = createCenturieRange(16, 20);
+
+export const PeopleFilters: React.FC<Props> = ({
+  people,
+  setCurrentPeopleList,
+  searchParams,
+  setSearchWith,
+}) => {
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  let newPeopleList: Person[] = people;
+
+  const sexKeys = Object.keys(Sex);
+  const sexValues = Object.values(Sex);
+
+  const createNewCenturieList = (newCenturie: string) => {
+    return centuries.includes(newCenturie)
+      ? centuries.filter(currentcenturie => currentcenturie !== newCenturie)
+      : [...centuries, newCenturie];
+  };
+
+  const findName = (name: string | null, normalizedQuery: string) => {
+    if (!name) {
+      return false;
+    }
+
+    return name.toLowerCase().includes(normalizedQuery);
+  };
+
+  const onNameFilter = () => {
+    const normalizedQuery = query.toLowerCase().trim();
+
+    if (normalizedQuery !== '') {
+      newPeopleList = newPeopleList.filter(
+        person =>
+          findName(person.name, normalizedQuery) ||
+          findName(person.fatherName, normalizedQuery) ||
+          findName(person.motherName, normalizedQuery),
+      );
+    }
+  };
+
+  const onSexFilter = () => {
+    if (sex !== '') {
+      newPeopleList = newPeopleList.filter(person => person.sex === sex);
+    }
+  };
+
+  const onCenturieFilter = () => {
+    const centuriesList = centuries.map(centurie => +centurie);
+
+    const firstPossibleYear = (centurie: number) => (centurie - 1) * 100 + 1;
+    const lastPossibleYear = (centurie: number) => centurie * 100 - 1;
+
+    const bornFilteredList: Person[] = [];
+
+    if (centuries.length > 0) {
+      for (const centurie of centuriesList) {
+        const thisCenturieList = newPeopleList.filter(
+          person =>
+            person.born >= firstPossibleYear(centurie) &&
+            person.born <= lastPossibleYear(centurie),
+        );
+
+        bornFilteredList.push(...thisCenturieList);
+      }
+
+      newPeopleList = bornFilteredList;
+    }
+  };
+
+  useEffect(() => {
+    onSexFilter();
+    onNameFilter();
+    onCenturieFilter();
+
+    setCurrentPeopleList(newPeopleList);
+  }, [sex, query, centuries]);
```

Comment:
Ensure that `setCurrentPeopleList` is included in the dependency array of `useEffect` if it is not guaranteed to be stable across renders. This can prevent potential bugs related to stale closures.